drone_flightplan/waypoints.py

```diff
@@ -2,345 +2,147 @@
 import argparse
 import pyproj
 import geojson
-from shapely.geometry import Polygon
-from drone_flightplan import calculate_parameters as cp
+from shapely.geometry import Point, shape
+from shapely.affinity import rotate
+from shapely.ops import transform
+from drone_flightplan.calculate_parameters import calculate_parameters as cp
 
-
-# Instantiate logger
 log = logging.getLogger(__name__)
 
 
-def generate_waypoints_within_polygon(
-    aoi: Polygon,
-    distance_between_lines_x: float,
-    distance_between_lines_y: float,
-    generate_each_points: bool,
-    generate_3d: bool,
-):
-    minx, miny, maxx, maxy = aoi.bounds
-
-    centroid = aoi.centroid
-
-    waypoints = [
-        {
-            "coordinates": (centroid.x, centroid.y),
-            "angle": "0",
-            "take_photo": False,
-            "gimbal_angle": "-90",
-        }
-    ]
-
-    # Generate waypoints within the polygon
-    y = miny
-    row_count = 0
-    angle = -90
+def generate_grid_in_aoi(aoi_polygon, x_spacing, y_spacing):
```

Comment:
Although not essential, type hinting is a really useful feature in Python - one of the understated advantages is to make it easier to understand code when looking back in the future!

```python
def generate_grid_in_aoi(aoi_polygon: shape, x_spacing: int, y_spacing: int):
```

This tells the current reviewer, future you, future collaborator that the aoi_polygon should be a shapely shape 👍

Of course, type hinting can also be used to validate params and do other things too!

Reply:
Appreciate the suggestion! I've added type hinting to all relevant functions in the latest commit.